modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ExtensionMethods.cs

```diff
@@ -165,6 +166,35 @@ public static string NameWithTypeParameters(this INamedTypeSymbol symbol)
         public static string FullQualifiedName(this INamespaceSymbol namespaceSymbol)
             => namespaceSymbol.ToDisplayString(FullyQualifiedFormatOmitGlobal);
 
+        public static string FullQualifiedName(this ISymbol symbol)
+            => symbol.ToDisplayString(FullyQualifiedFormatOmitGlobal);
+
+        public static string FullQualifiedSyntax(this SyntaxNode node, SemanticModel sm, bool isFirstNode = true)
+        {
+            if (node is NameSyntax ns && isFirstNode)
+            {
+                SymbolInfo nameInfo = sm.GetSymbolInfo(ns);
+                return nameInfo.Symbol?.FullQualifiedName() ?? ns.ToString();
+            }
+
+            StringBuilder sb = new StringBuilder(128);
+            bool innerIsFirstNode = true;
+            foreach (var child in node.ChildNodesAndTokens())
+            {
+                if (child.HasLeadingTrivia) { sb.Append(child.GetLeadingTrivia()); }
+
+                if (child.IsNode)
+                {
+                    sb.Append(child.AsNode().FullQualifiedSyntax(sm, isFirstNode: innerIsFirstNode));
```

Comment:
Do you think you could rewrite the `FullQualifiedSyntax` method as a private method that takes a `StringBuilder` and appends to it instead of returning a `string` to avoid allocating multiple strings? Then the public method would just create the `StringBuilder` and pass it to the private method.

Reply:
Done.